libs/parallelism/algorithms/include/hpx/parallel/algorithms/replace.hpp

```diff
@@ -652,9 +671,239 @@ namespace hpx { namespace parallel { inline namespace v1 {
 
         typedef hpx::is_sequenced_execution_policy<ExPolicy> is_seq;
 
-        return hpx::util::make_tagged_pair<tag::in, tag::out>(
-            detail::replace_copy_if<std::pair<FwdIter1, FwdIter2>>().call(
-                std::forward<ExPolicy>(policy), is_seq(), first, last, dest,
-                std::forward<F>(f), new_value, std::forward<Proj>(proj)));
+        return detail::replace_copy_if<
+            util::in_out_result<FwdIter1, FwdIter2>>()
+            .call(std::forward<ExPolicy>(policy), is_seq(), first, last, dest,
+                std::forward<F>(f), new_value, std::forward<Proj>(proj));
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+    ///////////////////////////////////////////////////////////////////////////
+    // CPO for hpx::replace_if
+    HPX_INLINE_CONSTEXPR_VARIABLE struct replace_if_t final
+      : hpx::functional::tag<replace_if_t>
```

Comment:
Could you please use `hpx::functional::tag_fallback` instead (see discussion in #5176)? The `tag_invoke` functions below would need to be renamed to `tag_fallback_invoke` as well.